hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/util/EnricherUtil.java

```diff
@@ -39,6 +39,24 @@ public static Map<String, AttributeValue> getAttributesForFirstExistingKey(
     return Collections.unmodifiableMap(attributes);
   }
 
+  public static Optional<org.hypertrace.core.datamodel.AttributeValue> getAttribute(
+      Attributes attributes, String key) {
+    return Optional.ofNullable(attributes)
```

Comment:
Use filter with findFirst

Reply:
Didn't get you, findFirst in the map?

Reply:
See my latest comment on this. You can stream the map entries, add a filter and then do a findFirst. This returns an Optional by default. Cleaner IMO but this is fine too.